test/e2e/gateway/007_gateway_allow_type_test.go

```diff
@@ -18,209 +18,206 @@ package gateway
-						{
-							Path: &gatewayv1alpha1.HTTPPathMatch{
-								Type:  pathMatchTypePtr(gatewayv1alpha1.PathMatchExact),
-								Value: stringPtr("/gateway-in-same-namespace"),
+			Spec: gatewayv1alpha1.HTTPRouteSpec{
+				Hostnames: []gatewayv1alpha1.Hostname{"gatewayallowtype.gateway.projectcontour.io"},
+				Gateways: &gatewayv1alpha1.RouteGateways{
+					Allow: gatewayAllowTypePtr(gatewayv1alpha1.GatewayAllowSameNamespace),
+				},
+				Rules: []gatewayv1alpha1.HTTPRouteRule{
+					{
+						Matches: []gatewayv1alpha1.HTTPRouteMatch{
+							{
+								Path: &gatewayv1alpha1.HTTPPathMatch{
+									Type:  pathMatchTypePtr(gatewayv1alpha1.PathMatchExact),
+									Value: stringPtr("/gateway-in-same-namespace"),
+								},
 							},
 						},
-					},
-					ForwardTo: []gatewayv1alpha1.HTTPRouteForwardTo{
-						{
-							ServiceName: stringPtr("echo"),
-							Port:        portNumPtr(80),
+						ForwardTo: []gatewayv1alpha1.HTTPRouteForwardTo{
+							{
+								ServiceName: stringPtr("echo"),
+								Port:        portNumPtr(80),
+							},
 						},
 					},
 				},
 			},
-		},
-	}
-	fx.CreateHTTPRouteAndWaitFor(gatewayInSameNamespaceRoute, httpRouteAdmitted)
-	defer func() {
-		require.NoError(t, fx.Client.Delete(context.TODO(), gatewayInSameNamespaceRoute))
-	}()
-
-	res, ok = fx.HTTP.RequestUntil(&e2e.HTTPRequestOpts{
-		Host:      string(gatewayInSameNamespaceRoute.Spec.Hostnames[0]),
-		Path:      "/gateway-in-same-namespace",
-		Condition: e2e.HasStatusCode(200),
-	})
-	require.Truef(t, ok, "expected 200 response code, got %d", res.StatusCode)
-
-	// This route allows gateways in the same namespace, and the actual
-	// gateway is *NOT* in the same namespace.
-	gatewayNotInSameNamespaceRoute := &gatewayv1alpha1.HTTPRoute{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      "gateway-not-in-same-namespace",
-			Labels:    map[string]string{"app": "filter"},
-		},
-		Spec: gatewayv1alpha1.HTTPRouteSpec{
-			Hostnames: []gatewayv1alpha1.Hostname{"gatewayallowtype.gateway.projectcontour.io"},
-			Gateways: &gatewayv1alpha1.RouteGateways{
-				Allow: gatewayAllowTypePtr(gatewayv1alpha1.GatewayAllowSameNamespace),
+		}
+		f.CreateHTTPRouteAndWaitFor(gatewayInSameNamespaceRoute, httpRouteAdmitted)
+
+		res, ok = f.HTTP.RequestUntil(&e2e.HTTPRequestOpts{
+			Host:      string(gatewayInSameNamespaceRoute.Spec.Hostnames[0]),
+			Path:      "/gateway-in-same-namespace",
+			Condition: e2e.HasStatusCode(200),
+		})
+		require.Truef(t, ok, "expected 200 response code, got %d", res.StatusCode)
+
+		// This route allows gateways in the same namespace, and the actual
+		// gateway is *NOT* in the same namespace.
+		f.CreateNamespace("007-gateway-allow-type-invalid")
+		defer f.DeleteNamespace("007-gateway-allow-type-invalid", false)
+		gatewayNotInSameNamespaceRoute := &gatewayv1alpha1.HTTPRoute{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "007-gateway-allow-type-invalid",
```

Comment:
Good example of where it'd be useful for the framework to support creating multiple namespaces for a test.